### PR TITLE
feat(textproto): implementation for all reader methods mirrored from the go standard library

### PR DESCRIPTION
**Pull request by load1n9**

### CODE REVIEW

**Comment by kt3k** [comment]
@load1n9 Thank you for the PR. These seem useful to me. Nice work! But we need to have test cases to accept the new features. Can you add test cases of added methods in `textproto/test.ts`?

**Comment by load1n9** [comment]
Sure

**Comment by load1n9** [comment]
added the tests

File: textproto/mod.ts [hunk 1]
```diff
@@ -142,6 +163,111 @@ export class TextProtoReader {
...
+    ) {
+      n--;
+    }
+    return s.slice(i, n);
+  }
+
+  async readCodeLine(
+    expectCode: number,
+  ): Promise<{ code: number; continued: boolean; message: string }> {
+    const line = await this.readCodeLine(expectCode);
```
Comment (kt3k) [inline textproto/mod.ts:244]: This seems like an infinite recursion?

File: textproto/mod.ts [hunk 2]
```diff
@@ -3,25 +3,42 @@
...
-
 const WHITESPACES: Array<number> = [CHAR_SPACE, CHAR_TAB];
-
 const decoder = new TextDecoder();
-
 // FROM https://github.com/denoland/deno/blob/b34628a26ab0187a827aa4ebe256e23178e25d39/cli/js/web/headers.ts#L9
 const invalidHeaderCharRegex = /[^\t\x20-\x7e\x80-\xff]/g;
 
-function str(buf: Uint8Array | null | undefined): string {
+export function str(buf: Uint8Array | null | undefined): string {
```
Comment (kt3k) [inline textproto/mod.ts:17]: Do we need to export this?

File: textproto/mod.ts [hunk 3]
```diff
@@ -3,25 +3,42 @@
...
   return !buf ? "" : decoder.decode(buf);
 }
+export function parseCodeLine(
```
Comment (kt3k) [inline textproto/mod.ts:20]: Ditto. Do we need to export this?

I don't find this function in go's textproto module

**Comment by kt3k** [comment]
Thanks for adding test cases, but I think we also need to test `readLineBytes` and `readContinuedLineSlice`

**Comment by load1n9** [comment]
Done

**Comment by kt3k** [comment]
Sorry for the delay in review. Do the test cases pass on your end? Some tests (`Reader: Code Reader` and `Reader: RFC959 Lines`) seem failing in the CI..

**Pull request closed without merging**